packages/block-library/src/query-pagination/edit.js

```diff
@@ -53,9 +54,11 @@ export default function QueryPaginationEdit( {
 		allowedBlocks: ALLOWED_BLOCKS,
 	} );
 	// Always show label text if paginationArrow is set to 'none'.
-	if ( paginationArrow === 'none' ) {
-		setAttributes( { showLabel: true } );
-	}
+	useEffect( () => {
+		if ( paginationArrow === 'none' ) {
```

Comment:
We don't need to set the state if `showLabel` is already `true`.
```suggestion
		if ( paginationArrow === 'none' && ! showLabel ) {
```